functional_tests/test_database_write.py

```diff
@@ -0,0 +1,162 @@
+            if p in self.dir_locks:
+                return self.dir_locks[p]
+            else:
+                new_lock = Lock()
+                self.dir_locks[p] = new_lock
+                return new_lock
+
+    def get_named_semaphore(self, lock_name, is_dir=False, count=1):
+        """
+        Returns a semaphore (which can be used as a context manager). The first time a lock_name
+        is received, a new semaphore will be established. Subsequent uses of that lock_name will
+        receive the same semaphore.
+        :param lock_name: A unique name for resource threads are contending over. If using a directory name
+                            as a lock_name, provide an absolute path.
+        :param is_dir: The lock_name is a directory (method will ignore things like trailing slashes)
+        :param count: The number of times the lock can be claimed. Default=1, which is a full mutex.
+        :return: A semaphore associated with the lock_name.
+        """
+        with self.named_semaphores['']:
+            if is_dir:
+                p = '_dir::' + str(pathlib.Path(str(lock_name)).absolute())  # normalize (e.g. strip trailing /)
+            else:
+                p = lock_name
+            if p in self.named_semaphores:
+                return self.named_semaphores[p]
+            else:
+                new_semaphore = Semaphore(count)
+                self.named_semaphores[p] = new_semaphore
+                return new_semaphore
+
+    @staticmethod
+    def timestamp():
+        return datetime.datetime.utcnow().isoformat()
+
+
+# Set up your local mariadb database before running this test
+class DBLibWriteTest(unittest.TestCase):
+    """
+    Doozer had been creating new column if a new label was added in a Dockerfile of an Openshift component. But since
+    the length of label names started to increase, doozer could not create a new column, as column names in SQL
+    can't be longer than 64 characters. The issue was mainly with labels which start with "io.openshift". So now only
+    the io.openshift labels in allow list are added as a column in the database. Other labels with length greater than
+    64 is also not inserted and is logged.
+
+    These tests will check if only the allowed labels values are inserted into the database.
+    """
+    def setUp(self) -> None:
+        self.fake_runtime = FakeRuntime()
+        self.db = DB(runtime=self.fake_runtime, environment="test")
+
+        self.db.connection = mysql.connector.connect(host='localhost',
+                                                     database='art_dash',
+                                                     user='root',
+                                                     password='secret')
+        self.db.mysql_db_env_var_setup = True  # since we're setting the db connection above
+        self.db.db = "art_dash"
+        self.table_name = "log_build"
+
+    def test_1(self):
```

Comment:
in general, good to have more descriptive test method names so that when it fails you recognize it...